### PR TITLE
libc/stdlib: Add implementation of random() and srandom()

### PR DESCRIPTION
**Pull request by lokeshbv**
Adds implementation of posix api's random() and srandom().

This patch is back-ported from Nuttx.

Signed-off-by: Lokesh B V <lokesh.bv@partner.samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cd1fba95551f9a9c249d341edf7c988fe719b91b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cd1fba95551f9a9c249d341edf7c988fe719b91b] - Code Rule Check OK.

File: os/include/stdlib.h [hunk 1]
```diff
@@ -188,6 +188,17 @@ void srand(unsigned int seed);
...
+#define srandom(s) srand(s)
+
+/**
+ * @ingroup STDLIB_LIBC
+ * @brief pseudo-random number generator
+ * @details @b #include <stdlib.h> \n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
+ * @since TizenRT v2.0 PRE
+ */
+long random(void);
```
[the remark this answers is not in the document]
Comment (lokeshbv) [reply os/include/stdlib.h]: 1. I will update tc for these new api's
2. Yes, it is standard api. Find it here, [stdlib.h](http://pubs.opengroup.org/onlinepubs/007904875/basedefs/stdlib.h.html)

**Comment by lokeshbv** [comment]
```
########## Kernel TC Start ##########

[tc_libc_stdlib_abort] PASS 

[tc_libc_stdlib_abs_labs_llabs] PASS 

[tc_libc_stdlib_atof] PASS 

[tc_libc_stdlib_atoi] PASS 

[tc_libc_stdlib_atol] PASS 

[tc_libc_stdlib_atoll] PASS 

[tc_libc_stdlib_bsearch] PASS 

[tc_libc_stdlib_div] PASS 

[tc_libc_stdlib_itoa] PASS 

[tc_libc_stdlib_qsort] PASS 

[tc_libc_stdlib_rand] PASS 

[tc_libc_stdlib_random] PASS <=== new tc
~~~~~~~~~~~~~~~~~~~~~~~~~~~

[tc_libc_stdlib_strtol] PASS 

[tc_libc_stdlib_strtoll] PASS 

[tc_libc_stdlib_strtoull] PASS 

[tc_libc_stdlib_strtoul] PASS 

[tc_libc_stdlib_strtod] PASS 

[tc_libc_stdlib_srand] PASS 

[tc_libc_stdlib_srandom] PASS <=== new tc
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

########## Kernel TC End [PASS : 19, FAIL : 0] ##########
```

**Comment by seinfra** [comment]
Target : [d0163a6231bb237dba07938d34c67b65bc176eaf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d0163a6231bb237dba07938d34c67b65bc176eaf] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_stdlib.c [hunk 2]
```diff
@@ -243,6 +243,34 @@ static void tc_libc_stdlib_rand(void)
...
+ * @brief                :random number generator
+ * @Scenario             :Returns a pseudo-random long integer number in the range between 0 and LONG_MAX.
+ * API's covered         :random
+ * Preconditions         :None
+ * Postconditions        :None
+ * @return               :void
+ */
+static void tc_libc_stdlib_random(void)
+{
+	long int ret_chk = ERROR;
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_stdlib.c]: It looks no need of initial value.

**Pull request merged**